plaso/parsers/text_plugins/snort_fastlog.py

```diff
@@ -0,0 +1,300 @@
+      r'(:\d*)?$')
+
+  _DATE_MONTH_DAY = (
+      text_parser.PyparsingConstants.TWO_DIGITS.setResultsName('month') +
+      pyparsing.Suppress('/') +
+      text_parser.PyparsingConstants.TWO_DIGITS.setResultsName('day_of_month'))
+
+  _DATE_YEAR_MONTH_DAY = (
+      text_parser.PyparsingConstants.TWO_DIGITS.setResultsName('year') +
+      pyparsing.Suppress('/') + _DATE_MONTH_DAY)
+
+  _DATE_TIME = (
+      (_DATE_YEAR_MONTH_DAY | _DATE_MONTH_DAY) +
+      pyparsing.Suppress(text_parser.PyparsingConstants.HYPHEN) +
+      text_parser.PyparsingConstants.TWO_DIGITS.setResultsName('hours') +
+      pyparsing.Suppress(':') +
+      text_parser.PyparsingConstants.TWO_DIGITS.setResultsName('minutes') +
+      pyparsing.Suppress(':') +
+      text_parser.PyparsingConstants.TWO_DIGITS.setResultsName('seconds') +
+      pyparsing.Suppress('.') +
+      pyparsing.Word(pyparsing.nums).setResultsName('fraction_of_second'))
+
+  _MESSAGE = pyparsing.Combine(pyparsing.OneOrMore(
+      pyparsing.Word(pyparsing.printables, excludeChars='["') |
+      pyparsing.White(' ', max=2))).setResultsName('message')
+
+  _RULE_IDENTIFIER = pyparsing.Combine(
+      text_parser.PyparsingConstants.INTEGER + ':' +
+      text_parser.PyparsingConstants.INTEGER + ':' +
+      text_parser.PyparsingConstants.INTEGER).setResultsName('rule_identifier')
+
+  _FASTLOG_LINE = (
+      _DATE_TIME + pyparsing.Suppress('[**] [') + _RULE_IDENTIFIER +
+      pyparsing.Suppress('] ') + pyparsing.Optional(pyparsing.Suppress('"')) +
+      _MESSAGE + pyparsing.Optional(pyparsing.Suppress('"')) +
+      pyparsing.Suppress('[**]') +
+      pyparsing.Optional(
+          pyparsing.Suppress(pyparsing.Literal('[Classification:')) +
+          pyparsing.Regex('[^]]*').setResultsName('classification') +
+          pyparsing.Suppress(']')) +
+      pyparsing.Optional(
+          pyparsing.Suppress('[Priority:') +
+          text_parser.PyparsingConstants.INTEGER.setResultsName('priority') +
+          pyparsing.Suppress(']')) +
+      pyparsing.Suppress('{') +
+      pyparsing.Word(pyparsing.alphanums).setResultsName('protocol') +
+      pyparsing.Suppress('}') +
+      text_parser.PyparsingConstants.IP_ADDRESS.setResultsName('src_ip') +
+      pyparsing.Optional(
+          pyparsing.Suppress(':') +
+          text_parser.PyparsingConstants.INTEGER.setResultsName('src_port')) +
+      pyparsing.Suppress('->') +
+      text_parser.PyparsingConstants.IP_ADDRESS.setResultsName('dst_ip') +
+      pyparsing.Optional(
+          pyparsing.Suppress(':') +
+          text_parser.PyparsingConstants.INTEGER.setResultsName('dst_port')) +
+      pyparsing.Suppress(pyparsing.lineEnd()))
+
+  _LINE_STRUCTURES = [('fastlog_line', _FASTLOG_LINE)]
+
+  _SUPPORTED_KEYS = frozenset([key for key, _ in _LINE_STRUCTURES])
+
+  def __init__(self):
+    """Initializes a parser."""
+    super(SnortFastLogTextPlugin, self).__init__()
+    self._last_month = 0
+    self._maximum_year = 0
```

Comment:
note to self _maximum_year and _plugin_by_reporter are not used, and to reset _year_use and _last_month on verification